python_basics_writing_python_code/python_basics_writing_python_code.md

```diff
@@ -0,0 +1,921 @@
+                      linkKey: "data"
+                      });       
+// Make *any* div with class 'python_data' a Sage cell
+sagecell.makeSagecell({inputLocation: 'div.python_data',
+                      evalButtonText: 'Run python',
+                      languages: ["python"],
+                      hide: ['fullScreen', 'permalink'],
+                      autoeval: 'false',
+                      linked: 'true',
+                      linkKey: "data"
+                      });                
+
+// Make *any* div with class 'r' a Sage cell
+sagecell.makeSagecell({inputLocation: 'div.r',
+                      evalButtonText: 'Run R',
+                      languages: ["r"],
+                      hide: ['fullScreen', 'permalink'],
+                      });
+// Make *any* div with class 'r_run' a Sage cell
+sagecell.makeSagecell({inputLocation: 'div.r_run',
+                      evalButtonText: 'Run R',
+                      languages: ["r"],
+                      hide: ['fullScreen', 'permalink'],
+                      autoeval: 'true'
+                      });
+</script>
+@end
+
+-->
+
+# Python Basics: Writing Python Code
+
+<div class = "overview">
+
+## Overview
+@comment
+
+**Is this module right for me?:** @long_description
+
+**Estimated time to completion:** @estimated_time
+
+**Pre-requisites:** Learners should be familiar with [Python as a programming language](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/update-intro-to-python/intro_to_python/intro_to_python.md#1), but need not have any experience with actually writing Python code.
+
+**Learning Objectives**:
+
+@learning_objectives
+
+
+</div>
+
+## Introduction to Python code
+
+Python is a flexible, multi-purpose programming language that has applications in web development, software engineering, and data analysis. Part of its power and much of its popularity comes from the fact that it is **human-readable**, with a relatively simple syntax compared to many other programming languages. However, just like any language, there are some grammar rules to learn in order to use Python. In this module, we'll go through a few of the basics of Python that will build a foundation for your learning and practice.
+
+### Functions and methods: Getting stuff done
+
+In Python, there are two basic ways of taking something (such as text, a number, or a sequence) and performing some action on it: functions and methods.
+
+Functions
+======
+
+**Functions** take the syntax `function(arguments)`: the name of the function, and then whatever you want to perform that function on (called **arguments**) in parentheses. A simple (but very useful) example of this is the `print()` function.
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+print("Hello World!")
+</script>
+</lia-keep>
+</div>
+
+The `print()` function takes an argument (in this case the phrase "Hello World!") and displays it as output.
```

Comment:
Maybe bold argument, or otherwise emphasize it? It isn't an unfamiliar word, but might be worth emphasizing here since people might not be as comfortable with this usage.

Reply:
It's bolded a few lines up when I define the term, and I don't want to get too bold-happy or it loses all meaning lol. But maybe this is a better spot?